demonstrations/tutorial_adaptive_circuits.py

```diff
@@ -0,0 +1,271 @@
+value of a molecular Hamiltonian with respect to all the existing the excitation gates and select
+gates based on the magnitude of the computed gradients. There are different ways to select the gates
+based on the computed gradients and here we discuss one of these strategies to compute the ground
+state energy of LiH. Both of these methods require constructing the Hamiltonian
+and determining all possible excitations which we can do with PennyLane functionalities shown below.
+But we first need to import the required libraries and define the molecular parameters including
+atomic symbols and coordinates. Note that the atomic coordinates are in Bohr.
+"""
+
+import pennylane as qml
+from pennylane import qchem
+from pennylane import numpy as np
+
+symbols = ["Li", "H"]
+geometry = np.array([0.000000000, 0.000000000, 0.000000000, 0.000000000, 0.000000000, 2.969280527])
+
+##############################################################################
+# We now compute the molecular Hamiltonian and the electronic excitations.
+
+H, qubits = qchem.molecular_hamiltonian(
+    symbols, geometry, charge=0, mult=1, basis="sto-3g", active_electrons=2, active_orbitals=5
+)
+
+singles, doubles = qchem.excitations(2, qubits)
+
+##############################################################################
+# Note that we have a total number of 24 excitation gates. We now implement a strategy which
+# constructs the circuit by adding groups of gate at a time. The first step is to compute the
+# gradients for a set of the existing gates. We create a circuit that applies a selected group of
+# gates to a reference Hartree-Fock state.
+
+
+def circuit_1(params, wires, excitations):
+    qml.BasisState(np.array([1, 1, 0, 0, 0, 0, 0, 0, 0, 0]), wires=wires)
+    for i, excitation in enumerate(excitations):
+        if len(excitation) == 4:
+            qml.DoubleExcitation(params[i], wires=excitation)
+        else:
+            qml.SingleExcitation(params[i], wires=excitation)
+
+
+##############################################################################
+# We construct our first group of gates by including all of the double excitations and compute the
+# gradients for each of the double excitation gates. We also need to define a
+# device and a cost function.
+
+dev = qml.device("default.qubit", wires=qubits)
+cost_fn = qml.ExpvalCost(circuit_1, H, dev, optimize=True)
+
+dcircuit = qml.grad(cost_fn, argnum=0)
+
+params = [0.0] * len(doubles)
+grads = dcircuit(params, excitations=doubles)
+
+print(grads)
+
+##############################################################################
+# The computed gradients have different values which reflect the contribution of each gate
+# in the final state prepared by the circuit. We select those gates that have a gradient above a
+# pre-defined threshold which we set to 0.001.
+
+doubles_select = [doubles[i] for i in range(len(doubles)) if abs(grads[i]) > 1.0e-5]

```

Comment:
```suggestion
doubles_select = [doubles[i] for i in range(len(doubles)) if abs(grads[i]) > 1.0e-3]
```
In the text you said threshold is 0.001